### PR TITLE
refactored recording & metronome for editor page

### PR DESCRIPTION
**Pull request by PLanzaS**
This was a pain in the ass to get to work because the way the video editor page was changed for the metronome completely fucks with the way I had it set up.

Should work, (I made sure to test it this time) but it is a bit janky.

### CODE REVIEW

File: src/renderer/pages/RecordingPage.vue [hunk 1]
```diff
@@ -254,29 +269,39 @@ export default defineComponent({
         })
       })
 
+    // Needs to wait until the metronomes are saved because they are created on the page switch
+    await new Promise(r => setTimeout(r, 500));
```
Comment (DaniFoldi) [inline src/renderer/pages/RecordingPage.vue:273]: This really is janky. I love it

**Comment by PLanzaS** [comment]
Upon further testing turns out that the recording wasn't working properly so I fixed it there

File: TestProject/.bones [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":1,"projectName":"TestProject","recordings":[],"options":{"audioTracks":[],"videoTracks":[]}}
```
Comment (DaniFoldi) [inline TestProject/.bones:1]: another test project 🎉

File: src/renderer/pages/RecordingPage.vue [hunk 3]
```diff
@@ -254,29 +269,39 @@ export default defineComponent({
...
+            playbackTracks.appendChild(header);
+          });
         });
       });
     });
 
     video.autoplay = true;
     this.startStreams();
-  }
+  },
```
Comment (DaniFoldi) [inline src/renderer/pages/RecordingPage.vue:304]: I love the overall code quality, it matches the repository greatly, could you run `npm run lint:fix` to make sure these are removed?

**Review by DaniFoldi — APPROVED** [review]
Please check the comments, otherwise lOoKsGoOdToMe

File: src/renderer/pages/RecordingPage.vue (hunk 3 above)
Comment (malachyoconnor) [reply src/renderer/pages/RecordingPage.vue:304]: hahaha

**Pull request merged**